server/src/main/java/io/druid/server/http/DatasourcesResource.java

```diff
@@ -551,8 +551,10 @@ public int compare(DruidDataSource druidDataSource, DruidDataSource druidDataSou
       }
 
       // segment stats
-      totalSegmentCount += druidDataSource.getSegments().size();
-      totalSegmentSize += dataSourceSegmentSize;
+      if (dataSourceSegmentSize >= totalSegmentSize) {
```

Comment:
Correct me if I'm wrong, but method **getSimpleDatasource** aggregates segment size/count only for one datasource on each server.